packages/debugger/src/panels/variables/mimerenderer.ts

```diff
@@ -19,30 +20,55 @@ export class VariableMimeRenderer extends MainAreaWidget<Panel> {
       content,
       reveal: Promise.all([dataLoader, loaded.promise])
     });
+    this.dataLoader = dataLoader;
+    this.renderMime = rendermime;
+    this._dataHash = null;
 
-    dataLoader
-      .then(async data => {
-        if (data.data) {
-          const mimeType = rendermime.preferredMimeType(data.data, 'any');
+    this.refresh()
+      .then(() => {
+        loaded.resolve();
+      })
+      .catch(reason => loaded.reject(reason));
+  }
+
+  /**
+   * Refresh the variable view
+   */
+  async refresh(): Promise<void> {
+    const data = await this.dataLoader();
 
-          if (mimeType) {
-            const widget = rendermime.createRenderer(mimeType);
-            const model = new MimeModel(data);
-            await widget.renderModel(model);
+    if (data.data) {
+      const hash = murmur2(JSON.stringify(data), 17);
+      if (this._dataHash !== hash) {
+        if (this.content.layout) {
+          this.content.widgets.forEach(w => {
+            this.content.layout?.removeWidget(w);
```

Comment:
`this.content.layout` nullish is already checked.

Reply:
This might indeed be improved in TS 4.4 (but may need to assign to a tmp variable to do the check): https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#cfa-aliased-conditions